### PR TITLE
Fix redundant fetch in zero copy replication

### PR DESCRIPTION
**Pull request by alesapin**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by alesapin** [comment]
How the dataloss looks like https://pastila.nl/?00238cc3/f4582d9737da16bf2b867e541d01f5cc

File: src/Storages/MergeTree/MergeFromLogEntryTask.cpp [hunk 1]
```diff
@@ -223,6 +223,27 @@ ReplicatedMergeMutateTaskBase::PrepareResult MergeFromLogEntryTask::prepare()
                     .part_log_writer = {}
                 };
             }
+            else if (!storage.findReplicaHavingCoveringPart(entry.new_part_name, true, dummy).empty())
```
Comment (tavplubix) [inline src/Storages/MergeTree/MergeFromLogEntryTask.cpp:226]:
```suggestion
            else if (!storage.findReplicaHavingCoveringPart(entry.new_part_name, /* active */ false, dummy).empty())
```

File: src/Storages/MergeTree/MutateFromLogEntryTask.cpp [hunk 2]
```diff
@@ -134,6 +134,27 @@ ReplicatedMergeMutateTaskBase::PrepareResult MutateFromLogEntryTask::prepare()
                     .part_log_writer = {}
                 };
             }
+            else if (!storage.findReplicaHavingCoveringPart(entry.new_part_name, true, dummy).empty())
```
Comment (tavplubix) [inline src/Storages/MergeTree/MutateFromLogEntryTask.cpp:137]:
```suggestion
            else if (!storage.findReplicaHavingCoveringPart(entry.new_part_name, /* active */ false, dummy).empty())
```

**Review by tavplubix — APPROVED** [review]

**Comment by tavplubix** [comment]
Also it worth adding a comment about the case with DROP_RANGE and SELECT on stale replica or server restart

**Comment by alesapin** [comment]
Failures unrelated and flaky on master.

**Pull request merged**